### PR TITLE
[metasrv] refine grpc

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* fix suggested changes in the previous PR #3606 
* move grpc to the independent crate.

## Changelog

- Improvement

## Related Issues

None

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/meta/grpc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+common-arrow = { path = "../../arrow" }
+common-exception = { path = "../../exception" }
+common-meta-api = { path = "../api" }
+common-meta-types = { path = "../types" }
+common-tracing = { path = "../../tracing" }
+common-grpc = {path = "../../grpc" }
```
Comment (drmingdrmer) [inline common/meta/grpc/Cargo.toml:18]: Sorted would always be better 🤔

File: common/meta/grpc/src/kv_api_impl.rs [hunk 2]
```diff
@@ -19,12 +19,12 @@ use common_meta_types::PrefixListReply;
...
-use crate::grpc::grpc_action::GetKVAction;
-use crate::grpc::grpc_action::MGetKVAction;
-use crate::grpc::grpc_action::PrefixListReq;
+use crate::grpc_action::GetKVAction;
+use crate::grpc_action::MGetKVAction;
+use crate::grpc_action::PrefixListReq;
 use crate::MetaGrpcClient;
 
-#[async_trait::async_trait]
+#[tonic::async_trait]
```
Comment (drmingdrmer) [inline common/meta/grpc/src/kv_api_impl.rs:27]: Why is it switched from `async_trait` to `tonic`?

**Review by drmingdrmer — APPROVED** [review]
Hm... appreciating these rational crate path in this patch!

File: common/meta/grpc/src/kv_api_impl.rs (hunk 2 above)
Comment (ariesdevil) [reply common/meta/grpc/src/kv_api_impl.rs:27]: `tonic::async_trait` is just re-pub `async_trait::async_trait`, using `tonic` reduce a dep crate:)

Comment (drmingdrmer) [reply common/meta/grpc/src/kv_api_impl.rs:27]: make sense 🤔

**Pull request merged**